src/briefcase/platforms/android/apk.py

```diff
@@ -1 +1,205 @@
-# An implementation would go here!
+import subprocess
+from pathlib import Path
+from zipfile import ZipFile
+
+from requests import exceptions as requests_exceptions
+
+# "Make adb a function call" should be an `integrations` wrapper like `xcode.py`
+
+from briefcase.commands import (
+    BuildCommand,
+    CreateCommand,
+    PackageCommand,
+    PublishCommand,
+    RunCommand,
+    UpdateCommand,
+)
+from briefcase.config import BaseConfig
+from briefcase.exceptions import BriefcaseCommandError, NetworkFailure
+
+
+class ApkMixin:
+    output_format = "apk"
+    platform = "android"
+
+    @property
+    def android_sdk_path_tmp(self):
+        # A path in which we prepare the Android SDK.
+        return Path.home() / ".briefcase" / "tools" / "android_sdk_tmp"
+
+    @property
+    def android_sdk_path(self):
+        return Path.home() / ".briefcase" / "tools" / "android_sdk"
+
+    def binary_path(self, app):
+        return (
+            self.platform_path
+            / app.formal_name
```

Comment:
BTW, why does "formal_name" exist? What's the real difference between an app name and its formal name? Anyway, here I'm hoping it's safe for the filesystem. I'm OK continuing with that assumption of filesystem-safe formal name, but it's a little odd.

Reply:
This is covered in the docs, but "Formal name" is a human readable name, and may include capitalization, punctuation, start with a number, or more. App name must be a PEP508 compliant name. Think "Hello World!" vs "hello_world".

We don't currently do any filesystem-safety checks on formal name - but you're right, we probably should. We *do* validate app_name is PEP508 compliant, though.

Reply:
+1

I'll continue writing code happily assuming it's filesystem-safe for now, then, and we can make a project-wide decision about that later/never. :)